modules/internal/DefaultRectangular.chpl

```diff
@@ -52,7 +52,7 @@ module DefaultRectangular {
   config param earlyShiftData = true;
   config param usePollyArrayIndex = false;
 
-  config param defaultRectangularSupportsAutoLocalAccess = false;
+  config param defaultRectangularSupportsAutoLocalAccess = true;
```

Comment:
What's the point of this `config param`? I can imagine it might have been useful when getting this working but I wouldn't expect it would make sense to toggle it off...

Reply:
This was initially added out of abundance of caution. An application loaded with foralls over default rectangular domains could see compilation time increases due to loop cloning. That loop cloning may not lead to any noticeable performance increase (particularly when `--auto-aggregation` is not used, which is default).

I haven't seen any pathological cases where something like that occur today, but to me, that possibility is still there.

Do you have significant concerns for keeping the flag in?

Reply:
Nope, I just didn't see the point of it. Presumably we are free to remove it in the future in a separate PR if everything is going well.

Reply:
I would be on board with that. Especially if it comes after some period of time where we don't see this to be a compilation speed issue.